cli/command/image/build.go

```diff
@@ -138,6 +141,9 @@ func (out *lastProgressOutput) WriteProgress(prog progress.Progress) error {
 	return out.output.WriteProgress(prog)
 }
 
+type translatorFunc func(context.Context, reference.NamedTagged) (reference.Canonical, error)
+type observerFunc func(context.Context, reference.Named)
```

Comment:
These are defined quite far from the single place they are used.  Please move these to just above where they are used.